skill-tree /src/main/java/com/RDS/skilltree/EndorsementList/EndorsementListDTO.java

```diff
@@ -0,0 +1,26 @@
+package com.RDS.skilltree.EndorsementList;
+
+import com.RDS.skilltree.User.UserModel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+
+import java.util.UUID;
+
+@Builder
+@Data
+@AllArgsConstructor
+public class EndorsementListDTO {
+    private UUID id;
+    private String description;
+    private UserModel endorser;
+    private EndorsementType type;
+
+    public static EndorsementListDTO toDTO(EndorsementListModel endorsementListModel) {
```

Comment:
in the toDTO() method. The method directly assigns the endorser field of the EndorsementListModel object to the endorser field of the newly constructed EndorsementListDTO object. This can lead to a memory leak or data corruption if the EndorsementListModel object is modified after it is converted to an EndorsementListDTO object.